### ISSUE TITLE
run rancher agent with userns=host when usernamespace is enabled

### ISSUE THREAD

**Issue by w3blogfr** [rancher/rancher#9597]
My docker is configured with users namespace.

To get privileged, we can add userns flag when run docker. Is disable for this container user namespace

`
docker run --userns=host --privileged

when I run docker-agent to add an host, the container launch multiple container agents (health, vps, scheduler, ...), but this option is not set in run.sh inside container

To solve the issue, I will have to build my own image.

It's good be a good solution to allow a extra flag in rancher-agent command, and add this to all docker run command executed inside.

**Comment by n-scope** [comment]
Users namespace is a great security feature introduced in docker 1.10.
Adding the extra flag --userns=host everywhere the --privileged flag would allow rancher users to use the namespace feature.

**Comment by Angelinsky7** [comment]
is there any news for this feature ?

**Comment by loganhz** [comment]
Thanks for your report!

With the release of Rancher 2.0, development on v1.6 is only limited to critical bug fixes and security patches.

If you think we should keep this issue open, please let me know.

**Comment by w3blogfr** [comment]
May be this ticket could be done for Rancher 2?

**Comment by stefanlasiewski** [comment]
Agreed. Allowing the use of Kernel User Namespaces in Rancher 2.x is a critical security feature, and is one of the top features recommended by several security vendors like Aqua as it helps to prevent root exploits.

**Comment by marriamda** [comment]
> My docker is configured with users namespace.
> 
> To get privileged, we can add userns flag when run docker. Is disable for this container user namespace
> 
> `
> docker run --userns=host --privileged
> 
> when I run docker-agent to add an host, the container launch multiple container agents (health, vps, scheduler, ...), but this option is not set in run.sh inside container
> 
> To solve the issue, I will have to build my own image.
> 
> It's good be a good solution to allow a extra flag in rancher-agent command, and add this to all docker run command executed inside.


Hi, I have the same mistake 
could you tell me what did you do with run.sh inside the container?

**Comment by w3blogfr** [comment]
in run.sh, anywhere you have docker run something, you ahve to add the "--userns=host --privileged" options, and should works.
